x/stakeibc/keeper/reward_converter_test.go

```diff
@@ -0,0 +1,425 @@
+		ConnectionId: withdrawalConnectionId,
+		Address: withdrawalAddress,
+	}
+
+	// Must initialize these or they will serialize differently from the default values
+	tradeConfig := types.TradeConfig{
+		SwapPrice: 				sdk.OneDec(),
+		MaxAllowedSwapLossRate: sdk.MustNewDecFromStr("0.05"),
+		MinSwapAmount: 			sdk.ZeroInt(),
+		MaxSwapAmount: 			sdk.NewIntFromUint64(uint64(1_000_000)),
+	}
+
+	// Create and set the trade route for testing
+	tradeRoute := types.TradeRoute{
+		RewardDenomOnHostZone: 	"ibc/reward_on_host",
+		HostAccount: 			hostICA,
+		TradeConfig: 			tradeConfig,
+	}
+	s.App.StakeibcKeeper.SetTradeRoute(s.Ctx, tradeRoute)
+
+	// Create and set the epoch tracker for timeouts
+	timeoutDuration := time.Second * 30
+	epochEndTime := uint64(s.Ctx.BlockTime().Add(timeoutDuration).UnixNano())
+	epochTracker := types.EpochTracker{
+		EpochIdentifier:    epochtypes.STRIDE_EPOCH,
+		NextEpochStartTime: epochEndTime,
+	}
+	s.App.StakeibcKeeper.SetEpochTracker(s.Ctx, epochTracker)
+
+	return tradeRoute
+}
+
+// Tests a successful WithdrawalRewardBalanceQuery
+func (s *KeeperTestSuite) TestWithdrawalRewardBalanceQuery_Successful() {
+	tradeRoute := s.SetupWithdrawalRewardBalanceQueryTestCase()
+
+	err := s.App.StakeibcKeeper.WithdrawalRewardBalanceQuery(s.Ctx, tradeRoute)
+	s.Require().NoError(err, "no error expected when querying balance")
+
+	// Check that one query was submitted
+	queries := s.App.InterchainqueryKeeper.AllQueries(s.Ctx)
+	s.Require().Len(queries, 1, "there should have been 1 query submitted")
+	query := queries[0]
+
+	// Confirm query contents
+	s.Require().Equal(tradeRoute.HostAccount.ChainId, query.ChainId, "query chain ID")
+	s.Require().Equal(tradeRoute.HostAccount.ConnectionId, query.ConnectionId, "query connection ID")
+	s.Require().Equal(icqtypes.BANK_STORE_QUERY_WITH_PROOF, query.QueryType, "query type")
+	s.Require().Equal(icqtypes.TimeoutPolicy_REJECT_QUERY_RESPONSE, query.TimeoutPolicy, "query timeout policy")
+	
+	// Confirm callback data
+	s.Require().Equal(types.ModuleName, query.CallbackModule, "query callback module")
+	s.Require().Equal(keeper.ICQCallbackID_WithdrawalRewardBalance, query.CallbackId, "query callback id")
+
+	var actualCallbackData types.TradeRoute
+	err = proto.Unmarshal(query.CallbackData, &actualCallbackData)
+	s.Require().NoError(err, "no error expected when unmarshalling callback data")
+
+	// Callback data should just be the trade route itself
+	s.Require().Equal(tradeRoute, actualCallbackData, "query callabck data")
+
+	// Confirm query request info
+	requestData := query.RequestData[1:] // Remove BalancePrefix byte
+	actualAddress, actualDenom, err := banktypes.AddressAndDenomFromBalancesStore(requestData)
+	s.Require().NoError(err, "no error expected when retrieving address and denom from store key")
+	s.Require().Equal(tradeRoute.HostAccount.Address, actualAddress.String(), "query account address")
+	s.Require().Equal(tradeRoute.RewardDenomOnHostZone, actualDenom, "query denom")
```

Comment:
I _think_ you could throw this in a helper function that takes a tradeRoute and query as args and does this validation, and then you could use it in the below tests